test/evp_kdf_test.c

```diff
@@ -1991,6 +1991,140 @@ static int test_kdf_ss_kmac(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+    static unsigned char EID1[] = {
+        0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+    static unsigned char password1[] = {'L', 'P', 'U', 'Q', 'D', 'L', 's', 'K' };
+
+    params = construct_snmpkdf_params("fail", EID1, sizeof(EID1),
+                                      (unsigned char *)password1, sizeof(password1));
+
+    /* Negative test - derive should fail */
+    ret = TEST_ptr(params)
+        && TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_SNMPKDF))
+        && TEST_int_eq(EVP_KDF_derive(kctx, out, sizeof(out), params), 0);
+
+    EVP_KDF_CTX_free(kctx);
+    OPENSSL_free(params);
+    return ret;
+}
+
+static int test_kdf_snmpkdf_zero_output_size(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+    static unsigned char EID1[] = {
+        0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+    static unsigned char password1[] = {'L', 'P', 'U', 'Q', 'D', 'L', 's', 'K' };
+
+    params = construct_snmpkdf_params("sha1", EID1, sizeof(EID1),
+                                      (unsigned char *)password1, sizeof(password1));
+
+    /* Negative test - derive should fail */
+    ret = TEST_ptr(params)
+        && TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_SNMPKDF))
+        && TEST_int_eq(EVP_KDF_derive(kctx, out, 0, NULL), 0);
+
+    EVP_KDF_CTX_free(kctx);
+    OPENSSL_free(params);
+    return ret;
+}
+
+static int test_kdf_snmpkdf_bad_pw(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+    static unsigned char EID1[] = {
+        0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+
+    params = construct_snmpkdf_params("sha1", EID1, sizeof(EID1),
+                                      NULL, 0);
+
+    /* Negative test - derive should fail */
+    ret = TEST_ptr(params)
+        && TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_SNMPKDF))
+        && TEST_int_eq(EVP_KDF_derive(kctx, out, sizeof(out), params), 0);
+
+    EVP_KDF_CTX_free(kctx);
+    OPENSSL_free(params);
+    return ret;
+}
+
+static int test_kdf_snmpkdf_bad_eid(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+    static unsigned char password1[] = {'L', 'P', 'U', 'Q', 'D', 'L', 's', 'K' };
+
+    params = construct_snmpkdf_params("sha1", NULL, 0,
+                                      (unsigned char *)password1, sizeof(password1));
+
+    ret = TEST_ptr(params)
+        && TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_SNMPKDF))
+        && TEST_int_eq(EVP_KDF_derive(kctx, out, sizeof(out), params), 0);
+
+    EVP_KDF_CTX_free(kctx);
+    OPENSSL_free(params);
+    return ret;
+}
+
+static int test_kdf_snmpkdf(void)
+{
```

Comment:
The tests here are to mainly used to cover corner cases not covered by evp_test. So potentially you could remove this one if it doesnt do much different from evp_test